public/app/core/angular_wrappers.ts

```diff
@@ -196,4 +197,11 @@ export function registerAngularDirectives() {
     ['onNowDelayChange', { watchDepth: 'reference', wrapApply: true }],
     ['onHideTimePickerChange', { watchDepth: 'reference', wrapApply: true }],
   ]);
+
+  react2AngularDirective('azureMonitorQueryEditor', QueryEditor, [
+    'query',
+    ['datasource', { watchDepth: 'reference' }],
+    'replaceTemplateVariable',
```

Comment:
replaceTemplateVariable is not part of the "real" react [QueryEditorProps](https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/types/datasource.ts#L343-L362) api, so I'd avoid using that. The datasource has access to the template variables, so I suggest you expose a function from the datasource that can expand template variables. This function could then be used from within the query editor components.